src/stories/Slider.stories.tsx

```diff
@@ -0,0 +1,56 @@
+import { Story } from '@storybook/react';
+import React from 'react';
+import { Box } from '@mui/material';
+import Slider, { SliderProps } from '../components/Slider';
+
+export default {
+  title: 'Slider',
+  component: Slider,
+};
+
+const Template: Story<SliderProps> = (args) => {
+  return <Box height='100vh' display='flex' alignItems='center'>
+    <Slider {...args} />
+  </Box>;
+};
+
+export const Default = Template.bind({});
+
+export const SnapToMark = Template.bind({});
+
+const marks = [
+  {
+    label: '0%',
+    value: 0,
+  },
+  {
+    value: 25,
+  },
+  {
+    label: '50%',
+    value: 50,
+  },
+  {
+    value: 75,
+  },
+  {
+    label: '100%',
+    value: 100,
+  },
+];
+
+Default.args = {
+  defaultValue: 50,
+  min: 0,
+  max: 100,
+  step: 1,
+  marks,
+  valueLabelDisplay: 'auto',
+};
+
+SnapToMark.args = {
+  defaultValue: 50,
+  marks,
+  valueLabelDisplay: 'auto',
+  onChange: (value) => alert(`Power level at ${value}%`)
```

Comment:
not sure why this is kicking in before the drag/snap

Reply:
Yeah, i noticed that, too, and wasn't really thrilled about it. I'll dig a little bit to see whether I can change that behavior

Reply:
ignore this comment.. looks like the drag is not animated, maybe mui behavior

Reply:
> Yeah, i noticed that, too, and wasn't really thrilled about it. I'll dig a little bit to see whether I can change that behavior

Sounds good, this seems like a good initial version maybe?

Reply:
I found a simple fix: using `onChangeCommitted` instead of `onChange` triggers on mouseup, meaning you can slide the slider around before getting the popup. I'll add the popup back in to that story to illustrate.

Reply:
Woot!

Reply:
Thanks for the callout here. I had just passed it over, but the solution was simple to find

Reply:
Sure 👍 glad they have a prop/callback. The only other thing is showing the drag before the snap, looks like mui has no out-of-box solution for that. We could explore later if Chudo wants, maybe.